### DISCUSSION TITLE
How to filter documents with multiple where conditions?

### DISCUSSION THREAD

**Discussion by ASMohamedFaheemAnver** [invertase/react-native-firebase#6216] [Q&A]
Let's say I have an object like this in Firestore,
```
[
  {
    users: ['randomUserOne', 'randomUserTwo'],
    messages: [],
    roomId: 'iIb7cKib6nyBSFBIW6w7',
  },
  {
    users: ['randomUserOne', 'randomUserThree'],
    messages: [],
    roomId: 'iIb7cKib6nyBSFBIW6w8',
  },
  {
    users: ['randomUserTwo', 'randomUserThree'],
    messages: [],
    roomId: 'iIb7cKib6nyBSFBIW6w9',
  },
];
```
I want to filter collections with users containing both 'randomUserOne' and 'randomUserTwo'.
According to my object I expect 
```
[
  {
    users: ['randomUserOne', 'randomUserTwo'],
    messages: [],
    roomId: 'iIb7cKib6nyBSFBIW6w7',
  }
];
```
I have tried something like this and it's giving me an error saying can't use 2 where close with 'array-contains' :'(

```
const db = firebase.firestore();
const chatsRef = db.collection('chats');
chatsRef
      .where('users', 'array-contains', 'randomUserOne')
      .where('users', 'array-contains', 'randomUserTwo')
```
Any idea?

**Comment by mikehardy** [comment]
Usage questions are by far best asked on stackoverflow, the official google team offers support there
Discussions here are best reserved for react-native-firebase specific questions
